### PR TITLE
Share Consent Screen for Debug Log (EXPOSUREAPP-5481)

### PR DESCRIPTION
**Pull request by axelherbstreith**
This PR adds the consent screen for sharing the debug log.

To test this PR, open "App-Informationen" -> "Fehlerberichte" -> Press "Starten" -> Press "Fehlerbericht senden"

image

### CODE REVIEW

**Review by janetback — APPROVED** [review]
UA reviewed

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/bugreporting/debuglog/ui/share/DebugLogShareFragment.kt [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+import de.rki.coronawarnapp.databinding.BugreportingDebuglogShareFragmentBinding
+import de.rki.coronawarnapp.util.di.AutoInject
+import de.rki.coronawarnapp.util.ui.observe2
+import de.rki.coronawarnapp.util.ui.popBackStack
+import de.rki.coronawarnapp.util.ui.viewBindingLazy
+import de.rki.coronawarnapp.util.viewmodel.CWAViewModelFactoryProvider
+import de.rki.coronawarnapp.util.viewmodel.cwaViewModels
+import javax.inject.Inject
+
+class DebugLogShareFragment : Fragment(R.layout.bugreporting_debuglog_share_fragment), AutoInject {
```
Comment (d4rken) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/bugreporting/debuglog/ui/share/DebugLogShareFragment.kt:17]: We called other classes "Upload", should we name this DebugLogUploadFragment then?

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/bugreporting/debuglog/ui/share/DebugLogShareFragment.kt [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+            debugLogSharePrivacyInformation.setOnClickListener {
+                // TODO Add navigation to new screen
+            }
+
+            toolbar.setNavigationOnClickListener { popBackStack() }
+        }
+
+        vm.routeToScreen.observe2(this) {
+            when (it) {
```
Comment (d4rken) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/bugreporting/debuglog/ui/share/DebugLogShareFragment.kt:39]: For simple things, we can just create a `routeToScreen = SingleLiveEvent<NavDirections>()` and skip the extra  navigation events class.

File: Corona-Warn-App/src/main/res/layout/bugreporting_debuglog_share_fragment.xml [hunk 3]
```diff
@@ -0,0 +1,158 @@
+<?xml version="1.0" encoding="utf-8"?>
+<layout xmlns:android="http://schemas.android.com/apk/res/android"
```
Comment (d4rken) [inline Corona-Warn-App/src/main/res/layout/bugreporting_debuglog_share_fragment.xml:2]: I don't think we need `<layout` here.

File: Corona-Warn-App/src/main/res/layout/include_debugging_debuglog_share_privacy_card.xml [hunk 4]
```diff
@@ -0,0 +1,115 @@
+<?xml version="1.0" encoding="utf-8"?>
+<layout xmlns:android="http://schemas.android.com/apk/res/android"
```
Comment (d4rken) [inline Corona-Warn-App/src/main/res/layout/include_debugging_debuglog_share_privacy_card.xml:2]: I don't think we need `<layout` here.

**Review by d4rken — COMMENTED** [review]
Looks good, few comments, no blockers.

**Comment by d4rken** [comment]
Rest of comments will be addressed in an final polishing PR.

**Pull request merged**